tests/suites/test_suite_cipher.function

```diff
@@ -1132,7 +1132,7 @@ exit:
 }
 /* END_CASE */
 
-/* BEGIN_CASE depends_on:MBEDTLS_CIPHER_AUTH_CRYPT */
+/* BEGIN_CASE depends_on:MBEDTLS_CIPHER_AUTH_CRYPT:MBEDTLS_CIPHER_MODE_AEAD */
```

Comment:
That doesn't look quite right to me. First, `AUTH_CRYPT` is defined as `AEAD || NIST_KW` it becomes redundant if we add `AEAD`. But more importantly, this function is mean to test NIST KW as well (it has specific code for that), so it should compile as soon NIST KW is enabled even if AEAD is not. Perhaps the AEAD-specific parts of the function should be guarded with `#if defined(MBEDTLS_CIPHER_MODE_AEAD)` but not the whole function.